DPL: add ability to disable inputs programmatically

ktf opened the pull request:
DPL: add ability to disable inputs programmatically

Disabled inputs will not result in an actual route for the data, however it
will be stored in the configuration, so that analysis workflows will not need
to have the configuration available at every step.

ktf commented:
@aalkin, as discussed, this should be what we need.

alibuild commented:
Error while checking build/O2/fullCI for 536881a3c9b82b4c3c1d7d46c8faa25bfb8a7fc7 at 2024-01-16 23:20:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2-full-system-test-latest/log
Detected critical problem in logfile digi.log
digi.log:[93641:internal-dpl-clock]: [FATAL] error while setting up workflow in o2-sim-digitizer-workflow: No string handling for argument 'bunchFilling' in state IN_DATAPROCESSOR
digi.log:[93642:SimReader]: [FATAL] error while setting up workflow in o2-sim-digitizer-workflow: No string handling for argument 'bunchFilling' in state IN_DATAPROCESSOR
digi.log:[93648:internal-dpl-ccdb-backend]: [FATAL] error while setting up workflow in o2-sim-digitizer-workflow: No string handling for argument 'bunchFilling' in state IN_DATAPROCESSOR
digi.log:[93661:TOFDigitizer]: [FATAL] error while setting up workflow in o2-sim-digitizer-workflow: No string handling for argument 'bunchFilling' in state IN_DATAPROCESSOR
[93641:internal-dpl-clock]: [FATAL] error while setting up workflow in o2-sim-digitizer-workflow: No string handling for argument 'bunchFilling' in state IN_DATAPROCESSOR
[93642:SimReader]: [FATAL] error while setting up workflow in o2-sim-digitizer-workflow: No string handling for argument 'bunchFilling' in state IN_DATAPROCESSOR
[93648:internal-dpl-ccdb-backend]: [FATAL] error while setting up workflow in o2-sim-digitizer-workflow: No string handling for argument 'bunchFilling' in state IN_DATAPROCESSOR
[93661:TOFDigitizer]: [FATAL] error while setting up workflow in o2-sim-digitizer-workflow: No string handling for argument 'bunchFilling' in state IN_DATAPROCESSOR


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/a6067244b3549e458a8d57ce3a65598c50ca5334/slc8_x86-64/o2checkcode/1.0-local24/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12537/536881a3c9b82b4c3c1d7d46c8faa25bfb8a7fc7/build_O2_fullCI/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI for 3cb857b0c7adcb96cd83aadc8d12df5488db1021 at 2024-01-28 07:14:
```
## sw/BUILD/O2-full-system-test-latest/log
task timeout reached .. killing all processes


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/36591d5a705691b08003d5578e5651e661b8ff53/slc8_x86-64/o2checkcode/1.0-local77/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12537/3cb857b0c7adcb96cd83aadc8d12df5488db1021/build_O2_fullCI/pretty.html).

ktf committed:
DPL: add ability to disable inputs programmatically

Disabled inputs will not result in an actual route for the data, however it
will be stored in the configuration, so that analysis workflows will not need
to have the configuration available at every step.

alibuild commented:
Error while checking build/O2/fullCI for 78f4a8d7f5a7663ac3ffb305eacd3c9716a919ca at 2024-02-10 07:35:
```
## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
/sw/SOURCES/O2/12537-slc8_x86-64/0/Framework/DataInspector/src/DataInspectorService.cxx:208:11: error: namespace 'o2::framework::DataInspector' does not follow the underscore convention [aliceO2-namespace-naming]
/sw/SOURCES/O2/12537-slc8_x86-64/0/Framework/DataInspector/src/DataInspectorService.cxx:241:36: error: namespace 'o2::framework::DataInspector' does not follow the underscore convention [aliceO2-namespace-naming]
/sw/SOURCES/O2/12537-slc8_x86-64/0/Framework/DataInspector/src/DataInspector.cxx:47:26: error: namespace 'o2::framework::DataInspector' does not follow the underscore convention [aliceO2-namespace-naming]
++ [[ 0 == 0 ]]
++ exit 1
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12537/78f4a8d7f5a7663ac3ffb305eacd3c9716a919ca/build_O2_fullCI/pretty.html).
